makefile

```diff
@@ -116,26 +116,27 @@ AR_is_$(subst $(space),_,$(origin AR)) := yes
 
 OSList          := lnx win mac
 
-o      := $(if $(OS_is_win),obj,o)
-a      := $(if $(OS_is_win),lib,a)
-d      := $(if $(OS_is_win),$(if $(MSVC_RT_is_debug),d,),)
-dtbb   := $(if $(OS_is_win),$(if $(MSVC_RT_is_debug),_debug,),)
-plib   := $(if $(OS_is_win),,lib)
-scr    := $(if $(OS_is_win),bat,sh)
-y      := $(notdir $(filter $(_OS)/%,lnx/so win/dll mac/dylib))
--Fo    := $(if $(OS_is_win),-Fo,-o)
--Q     := $(if $(OS_is_win),$(if $(COMPILER_is_vc),-,-Q),-)
--cxx17 := $(if $(COMPILER_is_vc),/std:c++17,$(-Q)std=c++17)
--fPIC  := $(if $(OS_is_win),,-fPIC)
+o           := $(if $(OS_is_win),obj,o)
+a           := $(if $(OS_is_win),lib,a)
+d           := $(if $(OS_is_win),$(if $(MSVC_RT_is_debug),d,),)
+dtbb        := $(if $(OS_is_win),$(if $(MSVC_RT_is_debug),_debug,),)
+plib        := $(if $(OS_is_win),,lib)
+scr         := $(if $(OS_is_win),bat,sh)
+y           := $(notdir $(filter $(_OS)/%,lnx/so win/dll mac/dylib))
+-Fo         := $(if $(OS_is_win),-Fo,-o)
+-Q          := $(if $(OS_is_win),$(if $(COMPILER_is_vc),-,-Q),-)
+-cxx17      := $(if $(COMPILER_is_vc),/std:c++17,$(-Q)std=c++17)
+-fPIC       := $(if $(OS_is_win),,-fPIC)
 -DMKL_ILP64 := $(if $(filter mkl,$(BACKEND_CONFIG)),-DMKL_ILP64)
--Zl    := $(-Zl.$(COMPILER))
--DEBC  := $(if $(REQDBG),$(-DEBC.$(COMPILER)) -DDEBUG_ASSERT -DONEDAL_ENABLE_ASSERT) -DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE
--DEBJ  := $(if $(REQDBG),-g,-g:none)
--DEBL  := $(if $(REQDBG),$(if $(OS_is_win),-debug,))
--EHsc  := $(if $(OS_is_win),-EHsc,)
--isystem := $(if $(OS_is_win),-I,-isystem)
--sGRP  = $(if $(OS_is_lnx),-Wl$(comma)--start-group,)
--eGRP  = $(if $(OS_is_lnx),-Wl$(comma)--end-group,)
+-Zl         := $(-Zl.$(COMPILER))
+-Zl_DPCPP   := $(-Zl.dpcpp)
+-DEBC       := $(if $(REQDBG),$(-DEBC.$(COMPILER)) -DDEBUG_ASSERT -DONEDAL_ENABLE_ASSERT) -DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE
+-DEBC_DPCPP := $(if $(REQDBG),$(-DEBC.dpcpp) -DDEBUG_ASSERT -DONEDAL_ENABLE_ASSERT) -DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE
```

Comment:
Not sure that `-DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE` are needed here.
Please check.
Other parts of the PR look good to me.